minkindr/include/kindr/minimal/implementation/quat-sim-transform-inl.h

```diff
@@ -0,0 +1,42 @@
+#ifndef KINDR_MINIMAL_IMPLEMENTATION_QUAT_SIM_TRANSFORM_INL_H_
+#define KINDR_MINIMAL_IMPLEMENTATION_QUAT_SIM_TRANSFORM_INL_H_
+
+namespace kindr {
+namespace minimal {
+
+template <typename Scalar>
+QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate() : scale_A_B_(1.) {}
+
+template <typename Scalar>
+QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate(
+    const Transform& T_A_B, const Scalar scale_A_B_)
+    : T_A_B_(T_A_B), scale_A_B_(scale_A_B_) {}
+
+template <typename Scalar>
+typename QuatSimTransformTemplate<Scalar>::Vectors
+QuatSimTransformTemplate<Scalar>::operator*(const Vectors& rhs) const {
+  return T_A_B_.transformVectorized(scale_A_B_ * rhs);
+}
+
+template <typename Scalar>
+QuatSimTransformTemplate<Scalar>
+QuatSimTransformTemplate<Scalar>::inverse() const {
+  return QuatSimTransformTemplate<Scalar>(
+      QuatTransformationTemplate<Scalar>(
+          T_A_B_.getRotation().inverse(), -T_A_B_.getRotation().inverseRotate(
+              T_A_B_.getPosition() / scale_A_B_)), 1. / scale_A_B_);
+}
+
+template<typename Scalar>
+std::ostream & operator<<(std::ostream & out,
+                          const QuatSimTransformTemplate<Scalar>& sim_3) {
+  out << "Transform:" << std::endl << sim_3.T_A_B_.getTransformationMatrix() <<
```

Comment:
Actually, a quick google reveals that there are two schools of when to apply scale: http://www.ethaneade.org/latex2html/lie/node28.html VS https://github.com/raulmur/ORB_SLAM2/blob/master/src/Sim3Solver.cc#L320 
Hmm, which one is correct?